crates/bevy_pbr/src/render/mesh_bindings.rs

```diff
@@ -1,229 +1,351 @@
 //! Bind group layout related definitions for the mesh pipeline.
+use std::{array, fmt};
 
+use bevy_asset::AssetId;
+use bevy_ecs::prelude::Resource;
 use bevy_math::Mat4;
-use bevy_render::{mesh::morph::MAX_MORPH_WEIGHTS, render_resource::*, renderer::RenderDevice};
+use bevy_render::mesh::{morph::MAX_MORPH_WEIGHTS, Mesh};
+use bevy_render::{render_resource::*, renderer::RenderDevice};
+use bevy_utils::HashMap;
+use bitflags::bitflags;
+use smallvec::SmallVec;
 
 use crate::render::skin::MAX_JOINTS;
+use crate::MeshUniform;
 
 const MORPH_WEIGHT_SIZE: usize = std::mem::size_of::<f32>();
 pub const MORPH_BUFFER_SIZE: usize = MAX_MORPH_WEIGHTS * MORPH_WEIGHT_SIZE;
 
 const JOINT_SIZE: usize = std::mem::size_of::<Mat4>();
 pub(crate) const JOINT_BUFFER_SIZE: usize = MAX_JOINTS * JOINT_SIZE;
 
-/// Individual layout entries.
-mod layout_entry {
-    use super::{JOINT_BUFFER_SIZE, MORPH_BUFFER_SIZE};
-    use crate::MeshUniform;
-    use bevy_render::{
-        render_resource::{
-            BindGroupLayoutEntry, BindingType, BufferBindingType, BufferSize, GpuArrayBuffer,
-            ShaderStages, TextureSampleType, TextureViewDimension,
+// --- Individual layout entries and individual bind group entries ---
+
+fn buffer_layout(binding: u32, size: u64, visibility: ShaderStages) -> BindGroupLayoutEntry {
```

Comment:
I think @IceSentry is doing something similar to simplify construction of bind group layout entries and bind group entries.